Emby.Naming/Video/StackResolver.cs

```diff
@@ -82,15 +73,20 @@ public IEnumerable<FileStack> ResolveAudioBooks(IEnumerable<AudioBookFileInfo> f
         /// Resolves videos from paths.
         /// </summary>
         /// <param name="files">List of paths.</param>
+        /// <param name="namingOptions">The naming options.</param>
         /// <returns>Enumerable <see cref="FileStack"/> of videos.</returns>
-        public IEnumerable<FileStack> Resolve(IEnumerable<FileSystemMetadata> files)
+        public static IEnumerable<FileStack> Resolve(IEnumerable<FileSystemMetadata> files, NamingOptions namingOptions)
         {
             var list = files
-                .Where(i => i.IsDirectory || VideoResolver.IsVideoFile(i.FullName, _options) || VideoResolver.IsStubFile(i.FullName, _options))
+                .Where(i => i.IsDirectory || VideoResolver.IsVideoFile(i.FullName, namingOptions) || VideoResolver.IsStubFile(i.FullName, namingOptions))
                 .OrderBy(i => i.FullName)
+                .Select(f => (f.IsDirectory, FileName: GetFileNameWithExtension(f), f.FullName))
                 .ToList();
 
-            var expressions = _options.VideoFileStackingRegexes;
+            // TODO is there a "nicer" way?
+            var cache = new Dictionary<(string, Regex, int), Match>();
```

Comment:
It's not super great, but the regex allocates a lot of strings (O(N^N))

Reply:
BenchmarkDotNet=v0.12.1, OS=Windows 10.0.19041.1348 (2004/?/20H1)
AMD Ryzen 5 3600, 1 CPU, 12 logical and 6 physical cores
.NET Core SDK=6.0.100
  [Host]     : .NET Core 6.0.0 (CoreCLR 6.0.21.52210, CoreFX 6.0.21.52210), X64 RyuJIT
  DefaultJob : .NET Core 6.0.0 (CoreCLR 6.0.21.52210, CoreFX 6.0.21.52210), X64 RyuJIT


|  Method |     Mean |   Error |  StdDev |     Gen 0 |    Gen 1 |    Gen 2 | Allocated |
|-------- |---------:|--------:|--------:|----------:|---------:|---------:|----------:|
|   Cache | 146.7 ms | 2.69 ms | 3.40 ms | 1000.0000 | 750.0000 | 250.0000 |   7.71 MB |
| NoCache | 142.0 ms | 1.91 ms | 1.70 ms | 1000.0000 |        - |        - |   9.83 MB |